index.ts

```diff
@@ -0,0 +1,476 @@
+	hasRecoil?: Recoil;
+	hasSecondaryEffect?: boolean;
+	heal?: number[] | null;
+	ignoreAbility?: boolean;
+	ignoreAccuracy?: boolean;
+	ignoreDefensive?: boolean;
+	ignoreEvasion?: boolean;
+	ignoreImmunity?: boolean | {[k: string]: boolean};
+	ignoreNegativeOffensive?: boolean;
+	ignoreOffensive?: boolean;
+	ignorePositiveDefensive?: boolean;
+	ignorePositiveEvasion?: boolean;
+	ignoresBurn?: boolean;
+	ignoresDefenseBoosts?: boolean;
+	infiltrates?: boolean;
+	isFutureMove?: boolean;
+	isSpread?: boolean | 'allAdjacent';
+	isZ?: boolean | ID;
+	multiaccuracy?: boolean;
+	multihit?: number | number[];
+	multihitType?: 'parentalbond';
+	noCopy?: boolean;
+	noDamageVariance?: boolean;
+	noFaint?: boolean;
+	noMetronome?: string[];
+	noPPBoosts?: boolean;
+	noSketch?: boolean;
+	nonGhostTarget?: string;
+	ohko?: true | 'Ice' | null;
+	percentHealed?: number;
+	pressureTarget?: string;
+	priority?: number;
+	pseudoWeather?: string:
+	recoil?: [number, number];
+	secondaries?: SecondaryEffect[];
+	self?: SelfEffect;
+	selfBoost?: {boosts?: Partial<BoostsTable>};
+	selfSwitch?: ID | boolean;
+	selfdestruct?: string | boolean;
+	sideCondition?: string;
+	sleepUsable?: boolean;
+	slotCondition?: string;
+	spreadHit: boolean;
+	spreadModifier?: number;
+	stallingMove?: boolean;
+	status?: StatusName;
+	stealsBoosts?: boolean;
+	terrain?: string;
+	thawsTarget?: boolean;
+	useSourceDefensive?: boolean;
+	useTargetOffensive?: boolean;
+	usesHighestAttackStat?: boolean;
+	volatileStatus?: ID;
+	weather?: ID;
+	willCrit?: boolean;
+	zMoveBoosts?: Partial<BoostsTable>;
+	zMoveEffect?: string;
+	zMovePower?: number;
+}
+
+class Moves {
+	static forGen(gen: Generation): Promise<Moves>;
+
+	get(name: string): Move | undefined;
+	getByID(id: ID): Move | undefined;
+	next(): Move | true;
+}
+
+// species.ts
+
+type GenderName = 'M' | 'F' | 'N';
+type UsageTierName = 'OU' | 'UU' | 'RU' | 'NU' | 'PU';
+type TierName = UsageTierName | 'Uber' | 'UUBL' | 'RUBL' | 'NUBL' | 'PUBL';
+type EvoType = 'trade' | 'stone' | 'levelMove' | 'levelExtra' | 'levelFriendship' | 'levelHold';
+interface SpeciesAbilities = {0: string, 1?: string, H?: string, S?: string};
+type MoveSource = string;
+
+interface EventInfo {
+	generation: Generation;
+	level?: number;
+	shiny?: boolean | 1;
+	gender?: GenderName;
+	nature?: NatureName;
+	ivs?: Partial<StatsTable>;
+	perfectIVs?: number;
+	isHidden?: boolean;
+	abilities?: ID[];
+	maxEggMoves?: number;
+	moves?: ID[];
+	pokeball?: ID;
+	from?: string;
+}
+
+interface Species extends Data {
+	type 'Species';
+
+	abilities: SpeciesAbilities;
+	addedType?: string;
+	baseForme?: string;
+	baseSpecies?: string;
+	baseStats: StatsTable;
+	battleOnly?: boolean
+	canHatch?: boolean;
+	color: string;
+	cosmeticForms?: ID[];
+	doublesTier?: TierName;
+	eggGroups: string[];
+	encounters?: EventInfo[]
+	eventOnly?: boolean
+	eventPokemon?: EventInfo[];
+	evoCondition?: string;
+	evoItem?: string;
+	evoLevel?: number;
+	evoMove?: string;
```

Comment:
would group all things related to an evolution into an object, which can be optional if there isn't an evolution